src/html/jsdoc.rs

```diff
@@ -193,8 +193,18 @@ pub fn markdown_to_html(
     #[cfg(feature = "tree-sitter")]
     ammonia_builder.add_allowed_classes("span", super::tree_sitter::CLASSES);
 
-    ammonia_builder.clean(&html).to_string()
-  };
+    html = ammonia_builder.clean(&html).to_string();
+  }
+
+  html.push_str(
+    r#"<script>
+for (const button of document.getElementsByTagName("button")) {
+  if (button.dataset["copy"]) {
+    button.onclick = () => navigator?.clipboard?.writeText(button.dataset["copy"]);
+  }
+}
```

Comment:
If I’m not wrong, this is what Luca exactly meant:

```
document.addEventListener("click", (e) => {
  if (e.target instanceof HTMButtonElement && e.target.dataset.copy) {
    return navigator?.clipboard?.writeText(e.target.dataset.copy);
  }
});
```

- It doesn’t override click listeners added by the user, unlike doing `.onclick = ...`.
- It doesn’t add a lot of click listeners.

Sorry for the interference.

Reply:
Yes that is what I meant.

Reply:
the reason i did the `onclick` is because this script will be present for each markdown thats rendered

Reply:
Is that more than one per page?

Reply:
yes. ie each example, each field of a class, each namespace member

Reply:
however the markdown rendering is the top-most common element. 
another solution would be having a separate template for this script that is added to pages, and JSR then needs to add that to the return payload as well

Reply:
Yeah I think that'd be better.